ee/modules/030-cloud-provider-vsphere/hooks/internal/v1/provider_cluster_configuration.go

```diff
@@ -0,0 +1,86 @@
+	InternalNetworkNames *[]string               `json:"internalNetworkNames,omitempty" yaml:"internalNetworkNames,omitempty"`
+	InternalNetworkCIDR  *string                 `json:"internalNetworkCIDR,omitempty" yaml:"internalNetworkCIDR,omitempty"`
+	VMFolderPath         *string                 `json:"vmFolderPath,omitempty" yaml:"vmFolderPath,omitempty"`
+	Region               *string                 `json:"region,omitempty" yaml:"region,omitempty"`
+	Zones                *[]string               `json:"zones,omitempty" yaml:"zones,omitempty"`
+	BaseResourcePool     *string                 `json:"baseResourcePool,omitempty" yaml:"baseResourcePool,omitempty"`
+	Layout               *string                 `json:"layout,omitempty" yaml:"layout,omitempty"`
+	Provider             *VsphereProvider        `json:"provider,omitempty" yaml:"provider,omitempty"`
+	Nsxt                 *VsphereNsxt            `json:"nsxt,omitempty" yaml:"nsxt,omitempty"`
+}
+
+type VsphereProvider struct {
+	Server   *string `json:"server,omitempty" yaml:"server,omitempty"`
+	Username *string `json:"username,omitempty" yaml:"username,omitempty"`
+	Password *string `json:"password,omitempty" yaml:"password,omitempty"`
+	Insecure *bool   `json:"insecure,omitempty" yaml:"insecure,omitempty"`
+}
+
+type VsphereMasterNodeGroup struct {
+	Replicas      *int32                         `json:"replicas,omitempty" yaml:"replicas,omitempty"`
+	Zones         *[]string                      `json:"zones,omitempty" yaml:"zones,omitempty"`
+	InstanceClass *VsphereNodeGroupInstanceClass `json:"instanceClass,omitempty" yaml:"instanceClass,omitempty"`
+}
+
+type VsphereNodeGroupInstanceClass struct {
+	VsphereInstanceClass
+	MainNetworkIPAddresses *[]VsphereMainNetworkIPAddresses `json:"mainNetworkIPAddresses,omitempty" yaml:"mainNetworkIPAddresses,omitempty"`
+}
+
+type VsphereMainNetworkIPAddresses struct {
+	Address     *string                 `json:"address,omitempty" yaml:"address,omitempty"`
+	Gateway     *string                 `json:"gateway,omitempty" yaml:"gateway,omitempty"`
+	Nameservers *IPAddressesNameservers `json:"nameservers,omitempty" yaml:"nameservers,omitempty"`
+}
+
+type IPAddressesNameservers struct {
+	Addresses *[]string `json:"addresses,omitempty" yaml:"addresses,omitempty"`
+	Search    *[]string `json:"search,omitempty" yaml:"search,omitempty"`
+}
+
+type VsphereNodeGroup struct {
+	Name          *string                        `json:"name,omitempty" yaml:"name,omitempty"`
+	Replicas      *int32                         `json:"replicas,omitempty" yaml:"replicas,omitempty"`
+	Zones         *[]string                      `json:"zones,omitempty" yaml:"zones,omitempty"`
+	NodeTemplate  *NodeTemplate                  `json:"nodeTemplate,omitempty" yaml:"nodeTemplate,omitempty"`
+	InstanceClass *VsphereNodeGroupInstanceClass `json:"instanceClass,omitempty" yaml:"instanceClass,omitempty"`
+}
+
+// NodeTemplate copied from 040-node-manager/hooks/internal/v1/nodegroup.go
```

Comment:
Aww, let's delete the following code.

Reply:
yep, we remove this code to pkg in node-manager module